src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -420,6 +421,7 @@ def get_objects_owned_by_application(self) -> List[ApplicationOwnedObject]:
             ).fetchall()
             return [{"name": row[1], "type": row[2]} for row in results]
 
+    @get_cli_context().metrics.start_span("create_or_upgrade_app")
```

Comment:
this will get the CLI context at module load time, which won't reflect forking. I think we need to always use the context manager form to make sure that we get the latest global context

Reply:
What about writing a custom decorator? I think that should grab the global context at runtime, and it avoids nesting the whole function, maybe something like:
```
def start_cli_metrics_span(span_name: str):
    def decorator(func):
        @wraps(func)
        def wrapper(*args, **kwargs):
            with get_cli_context().metrics.start_span(span_name):
                return func(*args, **kwargs)
        return wrapper
    return decorator
```

Reply:
yeah that would work